avatars/avatars.js

```diff
@@ -3328,13 +1762,129 @@ class Avatar {
+          }
+        };
+        const _processFingerBones = left => {
+          const fingerBones = left ? this.fingerBoneMap.left : this.fingerBoneMap.right;
+          const gamepadInput = left ? this.sdkInputs.leftGamepad : this.sdkInputs.rightGamepad;
+          for (const fingerBone of fingerBones) {
+            // if (fingerBone) {
+            const { bones, finger } = fingerBone;
+            let setter;
+            if (finger === 'thumb') {
+              setter = (q, i) => q.setFromAxisAngle(localVector.set(0, left ? -1 : 1, 0), gamepadInput.grip * Math.PI * (i === 0 ? 0.125 : 0.25));
+            } else if (finger === 'index') {
+              setter = (q, i) => q.setFromAxisAngle(localVector.set(0, 0, left ? 1 : -1), gamepadInput.pointer * Math.PI * 0.5);
+            } else {
+              setter = (q, i) => q.setFromAxisAngle(localVector.set(0, 0, left ? 1 : -1), gamepadInput.grip * Math.PI * 0.5);
+            }
+            for (let i = 0; i < bones.length; i++) {
+              setter(bones[i].quaternion, i);
+            }
+            // }
+          }
+        };
+        _processFingerBones(true);
+        _processFingerBones(false);
+      }
+    }
+    /* if (this.getTopEnabled() || this.getHandEnabled(0) || this.getHandEnabled(1)) {
+      _motionControls.call(this)
+    } */
+    
+    
+
+    _updateHmdPosition();
+    _applyAnimation(this, now, moveFactors);
+
+    if (this.poseAnimation) {
+      _overwritePose(this.poseAnimation);
+    }
+
+    // if (!this.getBottomEnabled()) {
+    localEuler.setFromQuaternion(this.inputs.hmd.quaternion, 'YXZ');
+    localEuler.x = 0;
+    localEuler.z = 0;
+    localEuler.y += Math.PI;
+    this.modelBoneOutputs.Root.quaternion.setFromEuler(localEuler);
+
+    this.modelBoneOutputs.Root.position.copy(this.inputs.hmd.position)
+      .sub(localVector.set(0, this.height, 0));
+    // }
+    /* if (!this.getTopEnabled() && this.debugMeshes) {
+      this.modelBoneOutputs.Hips.updateMatrixWorld();
+    } */
+
+
+    this.shoulderTransforms.Update();
+    this.legsManager.Update();
+
+    _updateEyeTarget();
+    _updateEyeballTarget();
+
+    this.modelBoneOutputs.Root.updateMatrixWorld();
+    Avatar.applyModelBoneOutputs(
+      this.foundModelBones,
+      this.modelBoneOutputs,
+      // this.getTopEnabled(),
+      this.getBottomEnabled(),
+      this.getHandEnabled(0),
+      this.getHandEnabled(1),
+    );
+    // this.modelBones.Root.updateMatrixWorld();
+
+
+    // this.springBoneTimeStep.update(timeDiff);
+    this.springBoneManager && this.springBoneManager.lateUpdate(timeDiffS);
+
+    // XXX hook these up
+    this.nodder.update(now);
+    this.emoter.update(now);
+    
+    this.options.visemes && _updateVisemes();
     _updateSubAvatars();
 
     if (metaversefile.isDebugMode() && !this.debugMesh) {
       this.debugMesh = _makeDebugMesh();
       this.debugMesh.wrapToAvatar(this);
       this.model.add(this.debugMesh);
     }
+    //redundant?
```

Comment:
It's not redundant, but probably dead code.

Reply:
yes, i wrote that while digging through things and just forgot to remove.  will remove in the update

Reply:
this comment has been removed